apps/notion/src/inngest/functions/users/sync-users.ts

```diff
@@ -0,0 +1,85 @@
+import type { User } from '@elba-security/sdk';
+import { eq } from 'drizzle-orm';
+import { NonRetriableError } from 'inngest';
+import { type NotionUser, deleteUsers, getUsers, updateUsers } from '@/connectors/users';
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { env } from '@/env';
+import { inngest } from '@/inngest/client';
+
+const formatElbaUser = (user: NotionUser): User => ({
+  id: user.id,
+  displayName: user.name,
+  email: user.person?.email,
+  additionalEmails: [],
+});
+
+type GetUserResult = {
+  next_cursor?: string;
+  results: [];
+};
+
+export const syncUsers = inngest.createFunction(
+  {
+    id: 'synchronize-users',
+    priority: { run: 'event.data.is_first_scan' ? '600' : '-600' },
+  },
+  { event: 'notion/users.sync.requested' },
+  async ({ event, step }) => {
+    const { organisationId, syncStartedAt, page } = event.data;
+
+    const {
+      ELBA_API_BASE_URL: integrationBaseUrl,
+      NOTION_VERSION: notionVersion,
+      NOTION_API_BASE_URL: sourceBaseUrl,
+      ELBA_SOURCE_ID: sourceId,
+      USERS_SYNC_JOB_BATCH_SIZE: pageSize,
+    } = env;
+
+    const token = await step.run('get-token', async () => {
+      const [organisation] = await db
+        .select({ token: Organisation.accessToken })
+        .from(Organisation)
+        .where(eq(Organisation.id, organisationId));
+
+      if (!organisation) {
+        throw new NonRetriableError(`Could not retrieve organisation with id=${organisationId}`);
+      }
+
+      return organisation.token;
+    });
+
+    const nextPage = await step.run('list-users', async () => {
+      const result = (await getUsers(
+        token,
+        pageSize,
+        sourceBaseUrl,
+        page,
+        notionVersion
+      )) as GetUserResult;
```

Comment:
```suggestion
      const result = await getUsers(
        token,
        pageSize,
        sourceBaseUrl,
        page,
        notionVersion
      );
```

The `GetUserResult` type is unnecessary, as `getUsers` is already typed correctly.